metrics/storage/webtooling.sh

```diff
@@ -0,0 +1,191 @@
+# https://github.com/v8/web-tooling-benchmark
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+TEST_NAME="web-tooling"
+IMAGE="docker.io/library/local-web-tooling:latest"
+DOCKERFILE="${SCRIPT_PATH}/web-tooling-dockerfile/Dockerfile"
+CI_JOB="${CI_JOB:-""}"
+configuration_file="/usr/share/defaults/kata-containers/configuration.toml"
+# Directory to run the test on
+# This is run inside of the container
+TESTDIR="${TESTDIR:-/testdir}"
+file_path="/web-tooling-benchmark"
+file_name="output"
+CMD="mkdir -p ${TESTDIR}; cd $file_path && node dist/cli.js > $file_name"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+NUM_CONTAINERS="$1"
+# This timeout is related with the amount of time that
+# webtool benchmark needs to run inside the container
+timeout=$((180 * "$NUM_CONTAINERS"))
+declare -a CONTAINERS_ID
+cpu_period="100000"
+cpu_quota="200000"
+
+TMP_DIR=$(mktemp --tmpdir -d webtool.XXXXXXXXXX)
+
+remove_tmp_dir() {
+	rm -rf "$TMP_DIR"
+}
+
+trap remove_tmp_dir EXIT
+
+# Show help about this script
+help(){
+cat << EOF
+Usage: $0 <count>
+   Description:
+        <count> : Number of containers to run.
+EOF
+}
+
+save_config(){
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"containers": "$NUM_CONTAINERS",
+		"image": "$IMAGE",
+		"units": "runs/s"
+	}
+EOF
+)"
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+}
+
+function main() {
+	# Verify enough arguments
+	if [ $# != 1 ]; then
+		echo >&2 "error: Not enough arguments [$@]"
+		help
+		exit 1
+	fi
+
+	containers=()
+	# Check tools/commands dependencies
+	cmds=()
+	cmds+=("docker")
+
+	setup_configuration_file
+	sudo systemctl restart containerd
+	init_env
+	check_cmds "${cmds[@]}"
+	check_ctr_images "$IMAGE" "$DOCKERFILE"
+
+	metrics_json_init
+
+	save_config
+
+	for ((i=1; i<= "$NUM_CONTAINERS"; i++)); do
+	containers+=($(random_name))
+		# Web tool benchmark needs 2 cpus to run completely in its cpu utilization
+		sudo -E ctr run -d --runtime io.containerd.run.kata.v2 --cpu-quota "${cpu_quota}" --cpu-period "${cpu_period}" "$IMAGE" "${containers[-1]}" sh -c "$PAYLOAD_ARGS"
+	done
+
+	# We verify that number of containers that we selected
+	# are running
+	for i in $(seq "$timeout") ; do
+		echo "Verify that the containers are running"
+		containers_launched=$(sudo ctr c list -q | wc -l)
+		[ "$containers_launched" -eq "$NUM_CONTAINERS" ] && break
+ 		sleep 1
+ 		[ "$i" == "$timeout" ] && return 1
+	done
+
+	# Launch webtooling benchmark
+ 	CONTAINERS_ID=($(sudo ctr c list -q))
+ 	for i in "${CONTAINERS_ID[@]}"; do
+		sudo ctr t exec -d --exec-id "$(random_name)" "$i" sh -c "$CMD"
+ 	done
+
+	# Verify that webtooling benchmark finished
+	CONTAINERS_ID=($(sudo ctr c list -q))
+	for j in $(seq 1 "$timeout"); do
+ 		FILE_CMD="cat $file_path/$file_name | grep Geometric"
+ 		check_file_content=$(sudo ctr t exec --exec-id "$(random_name)" "$i" sh -c "$FILE_CMD")
```

Comment:
you are using `$i` here, but in the `for` statement `j` is being used.

Reply:
changes applied